game-core/src/test/java/games/strategy/engine/data/changefactory/ObjectPropertyChangeMatcher.java

```diff
@@ -0,0 +1,34 @@
+package games.strategy.engine.data.changefactory;
+
+import games.strategy.engine.data.ChangeMatcher;
+import lombok.AllArgsConstructor;
+import org.hamcrest.Description;
+
+@AllArgsConstructor
+public class ObjectPropertyChangeMatcher extends ChangeMatcher<ObjectPropertyChange> {
+
+  private final String property;
+
+  private final Object newValue;
+
+  private final Object oldValue;
+
+  @Override
+  protected boolean matchesSafely(final ObjectPropertyChange item) {
+    return item.getProperty().equals(property)
+        && item.getNewValue().equals(newValue)
```

Comment:
Any nullity concerns here?

What if the new value or old value were null, and we were matching for a null value?